c/httpserver.c

```diff
@@ -3184,7 +3189,7 @@ static int handleHttpService(HttpServer *server,
         break;
       } /* else fall through */
     case SERVICE_AUTH_TOKEN_TYPE_LEGACY:
-      request->authenticated = serviceAuthNativeWithSessionToken(service,request,response,&clearSessionToken);
+      request->authenticated = serviceAuthNativeWithSessionToken(service,request,response,&clearSessionToken, &safRC, &racfRC, &racfRSN);
```

Comment:
This will need refactoring at some point. All those functions now have to have the safRC, racfRC, racfRSN parameters which don't directly relate to the functions.

We're leaking abstraction here, I don't think we should expose details like SAF, RACF. Perhaps returning some internal reason code and the translating it into the corresponding HTTP response would be better. This would also apply to the `safAuthenticate` call.